include/modules/xworkspaces.hpp

```diff
@@ -103,6 +105,7 @@ namespace modules {
     size_t m_index{0};
 
     event_timer m_timer{0L, 25L};
+    mutable std::mutex m_mutex;
```

Comment:
I renamed the mutex and added a comment. However, this one mutex is doing two things:
- serialize event-handling
- protect m_viewports

Is this double-usage of a mutex okay? The "mutable"-flag/modifier/info is only needed to protect the m_viewports, since the event-handler is const. Am I overthinking this?

Reply:
These 2 commits contain a lot of explanation on this:
* https://github.com/jaagr/polybar/commit/6c3881b18c01c820a13e3f92ed1121775d5dc1b0
* https://github.com/jaagr/polybar/commit/f4fab2ded14c2067c67e5f22cab1c05fe6b4f4a0